presto-hive/src/test/java/com/facebook/presto/hive/TestHiveSkipEmptyFiles.java

```diff
@@ -0,0 +1,269 @@
+                .put("hive.allow-drop-table", "true")
+                .put("hive.non-managed-table-writes-enabled", "true")
+                .put("hive.parquet.use-column-names", "true")
+                .put("hive.compression-codec", "GZIP")
+                .put("hive.storage-format", "PARQUET")
+                .put("hive.skip-empty-files", "false")
+                .build();
+        logger.info("  |-- Properties loaded");
+
+        logger.info("  |-- Creating catalog '" + CATALOG + "' using plugin '" + CATALOG + '\'');
+        this.queryFailRunner.createCatalog(CATALOG, CATALOG, properties);
+        logger.info("  |-- Catalog '" + CATALOG + "' created");
+        logger.info("  |-- Creating schema '" + SCHEMA + "' on catalog '" + CATALOG + '\'');
+        this.queryFailRunner.execute(format("CREATE SCHEMA %s.%s", CATALOG, SCHEMA));
+        logger.info("  |-- Schema '" + SCHEMA + "' created");
+
+        logger.info("'QueryFailRunner' created succesfully");
+    }
+
+    /**
+     * Generates a temporary directory and creates two parquet files inside, one is empty and the other is not
+     * @return a {@link File} pointing to the newly created temporary directory
+     */
+    private static File generateMetadata(String tableName)
+            throws Exception
+    {
+        // obtains the root resouce directory in order to create temporary tables
+        URL url = TestHiveSkipEmptyFiles.class.getClassLoader().getResource(".");
+        if (url == null) {
+            throw new RuntimeException("Could not obtain resource URL");
+        }
+        File temporaryDirectory = new File(url.getPath(), tableName);
+        boolean created = temporaryDirectory.mkdirs();
+        if (!created) {
+            throw new RuntimeException("Could not create resource directory: " + temporaryDirectory.getPath());
+        }
+        logger.info("Created temporary directory: " + temporaryDirectory.toPath());
+        File firstParquetFile = new File(temporaryDirectory, randomUUID().toString());
+        ParquetTester.writeParquetFileFromPresto(firstParquetFile,
+                ImmutableList.of(IntegerType.INTEGER),
+                Collections.singletonList("field"),
+                new Iterable[] {Collections.singleton(1)},
+                1,
+                GZIP,
+                PARQUET_2_0);
+        logger.info("First file written");
+        File secondParquetFile = new File(temporaryDirectory, randomUUID().toString());
+        if (!secondParquetFile.createNewFile()) {
+            throw new RuntimeException("Could not create empty file");
+        }
+        return temporaryDirectory;
+    }
+
+    /**
+     * Deletes the given directory and all of its contents recursively
+     * Does not follow symbolic links
+     * @param temporaryDirectory a {@link File} pointing to the directory to delete
+     */
+    private static void deleteMetadata(File temporaryDirectory)
+    {
+        File[] data = temporaryDirectory.listFiles();
```

Comment:
data --> files or metadataFiles or something like that

Reply:
Done